### ISSUE TITLE
Allowing plugins (or any external code) to override existing formats

### ISSUE THREAD

**Issue by seeschloss** [moment/moment#2165]
I am writing a library that handles French revolutionary calendar dates.

When looking into how I could use moment together with it, I found that it's quite difficult for a plugin to define new formats. The [jalaali plugin](/jalaali/moment-jalaali) has to completely wrap moment in order to add its _jYYYY_ and so on formats. In the process, it also has to reimplement (often just copying) internal functions, just because it can't access and modify moment's internal variables.

I tried a quick modification of moment's code to allow plugins to declare "extraFormats" that will then be handled just like the default internal ones, here it is: https://github.com/seeschloss/moment/commit/ff4920f9827af43b98bceccd2cc1c1a3be67eb69

That's not a pull request because:
1. I'm not sure the moment project is interested at all in the idea
2. It's far from complete, doesn't provide any support for parsing these extra formats
3. Kind of a quick hack (especially the really ugly RegExp modification part) and could be done in other, probably better, ways

Basically, before doing anything better I'd like to know what you think of this idea? Is it worth it?

**Comment by ichernev** [comment]
Adding formats, tokens etc is part of adding pluggable calendar systems: https://github.com/moment/moment/issues/1454

PRs welcome.

**Comment by marwahaha** [comment]
Closing due to age. Still, a cool idea, please open a PR if you're interested.
